server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2644,6 +2643,7 @@ private void verifyVmLimits(UserVmVO vmInstance, Map<String, String> details) {
             Map<String, String> customParameters = new HashMap<>();
             customParameters.put(VmDetailConstants.CPU_NUMBER, String.valueOf(newCpu));
             customParameters.put(VmDetailConstants.MEMORY, String.valueOf(newMemory));
+            customParameters.put(VmDetailConstants.CPU_SPEED, details.get(VmDetailConstants.CPU_SPEED));
```

Comment:
@shwstppr I see, there are resource limit checks for cpu count & memory (check below code), but not cpu speed. The new cpu speed changes are not recorded. Should this be addressed in another PR?

https://github.com/apache/cloudstack/blob/84b9b61b9ba079406e2c0c57e07153cff2a44b76/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L2656-L2657

https://github.com/apache/cloudstack/blob/84b9b61b9ba079406e2c0c57e07153cff2a44b76/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L2667-L2671